sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -186,8 +186,11 @@ def __init__(self, packages, options, environment_version, pipeline_url):
     if job_type.startswith('FNAPI_'):
       self.debug_options = self.debug_options or DebugOptions()
       self.debug_options.experiments = self.debug_options.experiments or []
+      # Don't add the default image overwrite if user overwrites or
```

Comment:
I think it will be easier to read if we restructure this:
```
if (self.debug_options.lookup_experiment('runner_harness_container_image') or 
    _use_unified_worker(self.debug_options)):
   # Comment on WHY we don't want to set the override
   pass
else:
   <...set the override...>
```
WDYT?

Reply:
Also, should we be passing pipeline_options instead of debug options? Perhaps it works either way but passing pipeline options would be cleaner since that's what the signature of _use_unified_worker() expects.

Reply:
> I think it will be easier to read if we restructure this:
> 
> ```
> if (self.debug_options.lookup_experiment('runner_harness_container_image') or 
>     _use_unified_worker(self.debug_options)):
>    # Comment on WHY we don't want to set the override
>    pass
> else:
>    <...set the override...>
> ```
> 
> WDYT?
I think it makes sense. Will update it.




> Also, should we be passing pipeline_options instead of debug options? Perhaps it works either way but passing pipeline options would be cleaner since that's what the signature of _use_unified_worker() expects.

I think the code is structured in a way that a `self.debug_options` is referenced every where and the reference to it can be different from `options` passed in the method argument. So I think it will need a more thoughtful refactoring to move to change it to options in the right manner. Which I think we can differ.